tests/testthat/test-cache.R

```diff
@@ -435,18 +411,14 @@ test_that("lint with cache uses the provided relative cache directory", {
 test_that("it works outside of a package", {
   linter <- assignment_linter()
 
-  with_mock(
-    `lintr::find_package` = function(...) NULL,
-    `lintr::pkg_name` = function(...) NULL,
-
-    path <- tempfile(pattern = "my_cache_dir_"),
-    expect_false(dir.exists(path)),
-    expect_lint("a <- 1", NULL, linter, cache = path),
-    expect_true(dir.exists(path)),
-    expect_length(list.files(path), 1),
-    expect_lint("a <- 1", NULL, linter, cache = path),
-    expect_true(dir.exists(path))
-  )
+  mockery::stub(lintr:::find_default_encoding, "find_package", function(...) NULL)
```

Comment:
Can we ensure this is the only call site that needs it?
Here are all usages of `find_package` in lintr:
![image](https://user-images.githubusercontent.com/2363178/157512913-ef655704-50bb-407a-93f0-206342ff1c40.png)

Reply:
my approach was: replace the `with_mock` function with `stop()` and look at the call stack. The public function call is `lint()` but `mockery::stub()` + `depth=` never seem to work for me, so I stubbed the private function instead. Tests are passing so that should be good?

Reply:
This test passing doesn't guarantee it passes for the right reason (i.e. in this context with no package to be found) IINM.
Can you check if this particular test passes if you manually replace the actual `find_package` function by `stop`?
e.g. something like 

```
devtools::load_all()
library(testthat)
environment(find_package)$find_package <- function(...) { message("Tripped!"); NULL }

# Test code
linter <- assignment_linter()

with_mock(
  `lintr::find_package` = function(...) NULL,
  `lintr::pkg_name` = function(...) NULL,

  path <- tempfile(pattern = "my_cache_dir_"),
  expect_false(dir.exists(path)),
  expect_lint("a <- 1", NULL, linter, cache = path),
  expect_true(dir.exists(path)),
  expect_length(list.files(path), 1),
  expect_lint("a <- 1", NULL, linter, cache = path),
  expect_true(dir.exists(path))
)
# > (no output)

# Verify trap works
expect_lint("a <- 1", NULL, linter, cache = path)
# > Tripped!
```

Reply:
I've gone through the full list now and I'm positive this was correct.

lint.R:273 is `lint_package()`- not called
lint.R:415 is `pkg_name()`, used by `namespace_linter()`, `object_name_linter()` and `object_usage_linter()` - but not by the `assignment_linter()` used for the test
namespace.R:2 is `namespace_linter()` - not called
object_name_linters.R:83 is `object_name_linter()` - not called
object_usage_linter.R:16 is `object_usage_linter()` - not called
settings.R:111 is `find_default_encoding()` - mocked correctly